packages/firebase_data_connect/firebase_data_connect/test/src/core/ref_test.dart

```diff
@@ -81,6 +89,141 @@ void main() {
       expect(queryManager.trackedQueries['testQuery']!['varsAsStr'], isNotNull);
       expect(stream, isA<Stream>());
     });
+
+    test('addQuery should return existing StreamController if query exists',
+        () {
+      final stream1 =
+          queryManager.addQuery('testQuery', 'variables', 'varsAsStr');
+      final stream2 =
+          queryManager.addQuery('testQuery', 'variables', 'varsAsStr');
+
+      expect(stream1, stream2);
+    });
+
+    test(
+      'subscribe should propagate errors via the stream',
+      () async {
+        final mockTransport = MockDataConnectTransport();
+        final queryManager = QueryManager(mockDataConnect);
+        final mockStreamController = MockStreamController<QueryResult>();
+        final completer = Completer<void>();
+
+        when(mockTransport.invokeQuery('testQuery', any, any, null))
```

Comment:
Is this needed since we trigger the error below?

Reply:
Indeed it's not needed, I've removed it